File: String/3174-minimum-number-of-changes-to-make-binary-string-beautiful/minimum-number-of-changes-to-make-binary-string-beautiful.py
class Solution:
    def minChanges(self, s: str) -> int:
        count = 0
        cur = s[0]
        total = 0

        for i in range(len(s)):
            if cur == s[i]:
                count += 1
            else:
                if count % 2 != 0:
                    total += 1
                    count = 0
                else:
                    count = 1
                cur = s[i]

        if count % 2 != 0:
            total += 1

        return total

File: String/3174-minimum-number-of-changes-to-make-binary-string-beautiful/minimum-number-of-changes-to-make-binary-string-beautiful.cpp
class Solution {
public:
    int minChanges(string s) {
        int count = 0;
        char cur = s[0];
        int total = 0;

        for(int i = 0; i < s.size(); i++){
            if(cur == s[i]) count++;
            else{
                if(count % 2 != 0){
                    total++;
                    count = 0;
                }
                else count = 1;
                cur = s[i];
            }
        }

        return total;
    }
};